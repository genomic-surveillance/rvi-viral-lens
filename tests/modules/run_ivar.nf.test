// Copyright (C) 2023 Genome Surveillance Unit/Genome Research Ltd.

nextflow_process {

    name "Test Process run_ivar"
    script "modules/run_ivar.nf"
    process "run_ivar"

    test("Should run without failures") {

        when {
            params {
                meta = [
                    id:"test2.9000000",
                    sample_id:"test2",
                    taxid:9000000
                ]
                test_loc="${projectDir}/tests/test_data"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0]=tuple(
                    params.meta,
                    file("${params.test_loc}/test_alignments/test*"),
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
