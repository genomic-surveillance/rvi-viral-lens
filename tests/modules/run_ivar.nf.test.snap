{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2.9000000",
                            "taxid": 9000000,
                            "sample_id": "test2"
                        },
                        [
                            "test2.9000000.sorted.bam.bai:md5,3246b03eab5b39f491e9159c2d5c583b",
                            "test.9000000.sorted.bam.bai:md5,536c3a693cabd649465bf8bd1fcd5aed",
                            "test.9000000.sorted.bam:md5,2f6b034826ae40672acd8059e084e788",
                            "test2.9000000.sorted.bam:md5,889f09bfe09b2d0d98c51fbbfd57dd3a"
                        ],
                        "test2.9000000.consensus.fa:md5,8fa86ad0980cad63ea403c60ac2a3fb3",
                        "test2.9000000_mutations.tsv:md5,c345f9265b8c12e27c8a9d3e8ed37814"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-24T18:02:03.971131"
    }
}